### ISSUE TITLE
Error running flake8

### ISSUE THREAD

**Issue by gtema** [AtomLinter/linter-flake8#582]
Since a while (~1 week) I get permanent problems with running flake8 inside of Atom:

```
linter-registry.js? [sm]:137 [Linter] Error running Flake8 Error: /bin/sh: module: line 1: syntax error: unexpected end of file
/bin/sh: error importing function definition for `module'
/bin/sh: ml: line 1: syntax error: unexpected end of file
/bin/sh: error importing function definition for `ml'
    at ChildProcess.<anonymous> (/home/gtema/.atom/packages/linter-flake8/node_modules/sb-exec/lib/index.js:56)
    at emitTwo (events.js:106)
    at ChildProcess.emit (events.js:194)
    at maybeClose (internal/child_process.js:899)
    at Socket.<anonymous> (internal/child_process.js:342)
    at emitOne (events.js:96)
    at Socket.emit (events.js:191)
    at Pipe._handle.close [as _onclose] (net.js:510)
```

I have tried reinstalling linter, linter-flake8 couple of times with no luck, so I need to keep it disabled.

Flake executable is present and working by itself correct.

**Comment by Arcanemagus** [comment]
That error is coming from trying to run `flake8`, are you able to run it manually?

**Comment by gtema** [comment]
Yes, it is running perfectly manually. It finds errors, but doesn't crash.
There are no such errors, only regular syntax. And with those errors it was
working correctly.

**Comment by gtema** [comment]
Done this already, doesn't help. Deleted linter-flake8 and linter
completely.

**Comment by gtema** [comment]
reinstalled Atom completely (removing ~/.atom) and still the same issue on ANY python project.
manual invocation of python3-flake8 (as also configured in linter-flake8) is passing without problems.
For the reference, after installation of the atom no other additional packages except of all dependencies of linter-flake8 were installed.
Please advice if any debug possibility exist.

**Comment by MRossol** [comment]
I was having the same issue after recently updating my python packages. Running: python -m pip install flake8 fixed the issue for me

**Comment by gtema** [comment]
No, it is not that. Flake is installed and working properly. What is/was
wrong is that 'lua' sets bash functions "ml" and "module". Some of the atom
packages (including new terminal from atom-ide) are getting crazy with it.
Uninstalling lua (or ensuring those bash functions are missing resolves the
issue, but one need to look why is this happening at first place.

Regards

**Comment by Arcanemagus** [comment]
Can you run `npm ls consistent-env` in the `linter-flake8` package's folder?

```
.atom\packages\linter-flake8> npm ls consistent-env
linter-flake8@2.3.0 C:\Users\abneyl\.atom\packages\linter-flake8
+-- atom-linter@10.0.0
| `-- sb-exec@4.0.0
|   `-- consistent-env@1.3.1
`-- atom-package-deps@4.6.0
  `-- sb-exec@3.1.0
    `-- consistent-env@1.3.1  deduped
```

As far as I understand this _should_ have been fixed in https://github.com/steelbrain/consistent-env/pull/40, one could also argue that putting newlines in environment variables is pure insanity, but there are a few system packages on some distros that seem to like watching the world burn.

**Comment by nameforaperson** [comment]
Having a similiar issue, when linter8 is trying to run, it is asking to run console to view errors, in console it shows the following
C:\Users\username\.atom\packages\linter\lib\linter-registry.js:159 [Linter] Error running Flake8 Error: c:\users\username\appdata\local\programs\python\python37\lib\site-packages\pycodestyle.py:113: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
    at ChildProcess.<anonymous> (C:\Users\username\.atom\packages\linter-flake8\node_modules\sb-exec\lib\index.js:56:20)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
(anonymous) @ C:\Users\username\.atom\packages\linter\lib\linter-registry.js:159
C:\Users\username\.atom\packages\linter\lib\linter-registry.js:159 [Linter] Error running Flake8 Error: c:\users\username\appdata\local\programs\python\python37\lib\site-packages\pycodestyle.py:113: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
    at ChildProcess.<anonymous> (C:\Users\username\.atom\packages\linter-flake8\node_modules\sb-exec\lib\index.js:56:20)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:925:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)



and when running flake8 --version in cmd, getting

c:\users\username\appdata\local\programs\python\python37\lib\site-packages\pycodestyle.py:113: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
3.5.0 (mccabe: 0.6.1, pycodestyle: 2.3.1, pyflakes: 1.6.0) CPython 3.7.0 on Windows


I am fairly new to programming, running a new install of atom and trying to use flake8 with the plugin, have restarted atom and installed all flake8 depencies. Tried the npm ls consistent-env line in the packages directory but don't have npm - looks like it's for javascript? Could anyone give advice on a fix, would like to use this linter as it looks helpful for debugging and I expect to be doing a lot of that :)

**Comment by malkir23** [comment]
It's because python 3.7 causes an error for old syntax. So you can wait till it will be fixed or install python 3.6.6

**Comment by Arcanemagus** [comment]
The original reporter's bug should have been fixed in one of the dependencies. The subsequent (and unrelated) error discussion in here is resolved by the comment above mine.

Thanks for taking the time to file an issue!
